packages/components/src/DateInput.tsx

```diff
@@ -12,7 +12,7 @@ const DATE_FORMAT = 'YYYY-MM-DD';
 
 type DateInputProps = {
   className?: string;
-  onChange?: (date?: string) => void;
+  onChange: (date: string) => void; // Remove the question mark to make date non-optional
```

Comment:
No need to add this comment here. 
```suggestion
  onChange: (date: string) => void;
```